src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,110 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> entering;
+    private int size;
+
+    public MyLinkedList() {
+        entering = new Node<>(null, null, null);
+        entering.next = entering.previous = entering;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> tail = index == size ? entering : getNode(index);
+        Node<T> element = new Node<>(value, tail, tail.previous);
+        element.previous.next = element.next.previous = element;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element, size);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> element = getNode(index);
+        T prevValue = element.value;
+        element.value = value;
+        return prevValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
```

Comment:
checklist: "Create a separate method unlink(Node node).
This logic will be used in 2 methods: when we remove from list by index and by value. So let’s create a method that will take a Node that needs to be removed and unlink it. As a result we will call it from both methods when Node is found."